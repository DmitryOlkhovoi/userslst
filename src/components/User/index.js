import _ from 'underscore'
import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { Card, CardBody, CardTitle, CardSubtitle, CardText, Button } from 'reactstrap'
import './styles/index.css'

class User extends Component {
  render() {
    const { id, name, email, text } = this.props.data

    return (
      <Card className="user">
        <CardBody>
          <CardTitle>{name}</CardTitle>
          <CardSubtitle>{email}</CardSubtitle>
          <CardText>{text}</CardText>
          <Link to={`/edit/${id}`}>
            <Button>Update</Button>
          </Link>
          <Button color="danger" onClick={() => this.props.onDelete(id)}>Delete</Button>
        </CardBody>
      </Card>
    )
  }
}

User.propTypes = {
  onDelete: PropTypes.func,
  data: PropTypes.object,
}

User.defaultProps = {
  onDelete: _.noop,
  data: {
    id: ':id',
    name: ':name',
    email: ':email',
    text: ':text',
  },
}

export default User
