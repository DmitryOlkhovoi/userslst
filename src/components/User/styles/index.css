.user {
  margin-top: 25px;
}

button.btn {
  margin: 5px;
}