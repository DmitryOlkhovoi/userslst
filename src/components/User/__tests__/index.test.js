import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import User from '../index'

it('renders correctly', () => {
  const tree = renderer.create(
  <MemoryRouter><User /></MemoryRouter>
).toJSON()
  expect(tree).toMatchSnapshot()
})
