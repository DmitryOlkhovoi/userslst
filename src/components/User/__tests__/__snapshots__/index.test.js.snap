// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="user card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      :name
    </h5>
    <h6
      className="card-subtitle"
    >
      :email
    </h6>
    <p
      className="card-text"
    >
      :text
    </p>
    <a
      href="/edit/:id"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary"
        onClick={[Function]}
        type={undefined}
      >
        Update
      </button>
    </a>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
